src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -281,10 +292,17 @@ export function getMostRecentBundleBlockRanges(
         }: ${JSON.stringify(rootBundleBlockRanges)}`
       );
     }
-    const blockRangeForChain = getBlockRangeForChain(rootBundleBlockRanges, chainId);
 
-    // Push the structure to the start of the list
-    bundleData.unshift({
+    // If UBA is not enabled for this bundle, exit early since no subsequent bundles will be enabled
+    // for the UBA, as the UBA was a non-reversible change.
+    const hubPoolStartBlock = getBlockRangeForChain(rootBundleBlockRanges, hubPoolClient.chainId)[0];
+    if (!isUbaBlock(hubPoolStartBlock, hubPoolClient.configStoreClient)) {
```

Comment:
Would we want to be checking for a specific spoke or is the hubpool block the one we're looking to verify?

Reply:
I think we need to always compare the hub pool start block against the config store blocks since they're both on mainnet